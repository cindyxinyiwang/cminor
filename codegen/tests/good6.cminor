// for statement
y: integer;

func: function integer(i:integer, j:integer) = {
	return i+j;
}

printarray: function void ( a: array [] integer, size: integer ) = {
	i: integer;
	for( i=0;i<size;i++) {
		print a[i], "\n";
	}
	x: integer;
	for (x=12; x< func(2,3); x++) 
	for(y=0; y<13; y--)
	x++;
	return ;
}