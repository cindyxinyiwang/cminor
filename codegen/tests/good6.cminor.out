i resovles to param 0
j resovles to param 1
i resovles to local 0
i resovles to local 0
size resovles to param 1
i resovles to local 0
a resovles to param 0
i resovles to local 0
x resovles to local 1
x resovles to local 1
func resovles to global func
x resovles to local 1
y resovles to global y
y resovles to global y
y resovles to global y
x resovles to local 1
